lib/turn.rb

```diff
@@ -0,0 +1,73 @@
+class Turn 
+  attr_reader :player1, 
+              :player2, 
+              :spoils_of_war 
+
+  def initialize(player1, player2)
+    @player1 = player1
+    @player2 = player2 
+    @spoils_of_war = []
+  end
+
+  def type
+    if (@player1.deck.rank_of_card_at(0) == @player2.deck.rank_of_card_at(0)) &&
+      ((@player1.deck.rank_of_card_at(2) == @player2.deck.rank_of_card_at(2)))
+      :mutually_assured_destruction
+    elsif @player1.deck.rank_of_card_at(0) == @player2.deck.rank_of_card_at(0)
+      :war
+    else
+      :basic
+    end
+  end
+
+  def winner 
+    if type == :mutually_assured_destruction
+      puts "No Winner"
+    elsif type == :war
+      if player1.deck.rank_of_card_at(2) > player2.deck.rank_of_card_at(2)
+        player1
+      elsif player2.deck.rank_of_card_at(2) > player1.deck.rank_of_card_at(2)
+        player2
+      end
+    else
+      if @player1.deck.rank_of_card_at(0) > @player2.deck.rank_of_card_at(0)
+        return @player1
+      else
+        return @player2
+      end
+    end
+  end
+
+  def pile_cards
+    if type == :mutually_assured_destruction
+      @player1.deck.cards.shift(3)
+      @player2.deck.cards.shift(3)
+    elsif type == :war
+      @spoils_of_war << @player1.deck.cards.shift(3)
+      @spoils_of_war << @player2.deck.cards.shift(3)
+      @spoils_of_war.flatten! 
+    else type == :basic 
+      @spoils_of_war << @player1.deck.cards.shift()
+      @spoils_of_war << @player2.deck.cards.shift()
+    end
+  end
+
+  def award_spoils(winner)
+    @spoils_of_war.each do |card|
+      winner.deck.cards << card
+    end
+    @spoils_of_war = []
+  end
```

Comment:
This method is wonderfully written and works as intended if the game's are of a `:basic` or `:war` type. If the game is `:mutually_assured_destruction` the game breaks here. This is because, if you remember, when a game is `:mutually_assured_destruction` we get `"No Winner"` returned from the `winner` method. And therefore when we call `award_spoils` and pass it `winner`, it will have the value `"No Winner"` string and you cannot call `.deck` on a String.

One way to avoid this behavior is to add a conditional statement to this method so that way it only runs if an actual player object is stored in `winner`.

```ruby
def award_spoils(winner)
  if winner != "No Winner" #<== If winner is not equal to "No Winner"
    @spoils_of_war.each do |card| #<== then do this
      winner.deck.cards << card
    end
  end
  @spoils_of_war = []
end
```